lib/HALSim/src/Network.cpp

```diff
@@ -203,8 +212,17 @@ bool Network::connect()
     {
         LOG_DEBUG("MQTT client connected to broker");
         resubscribe();
-        m_state   = STATE_CONNECTED;
-        isSuccess = true;
+        m_state = STATE_CONNECTED;

```

Comment:
Is the mosquitto_connect() blocking until the connection is established?

Reply:
It is not blocking but it is also not async. The function returns an error if not able to connect, without waiting. There is also an async method but requires refactoring of the loop.